backend/app/api/endpoints/knowledge.py

```
@@ -262,9 +262,16 @@ def get_knowledge_base(
             knowledge_base_id=knowledge_base_id,
         )
     except ValueError as e:
+        error_msg = str(e)
+        # Check if it's an access denied error
+        if "access denied" in error_msg.lower():
+            raise HTTPException(
+                status_code=status.HTTP_403_FORBIDDEN,
+                detail=error_msg,
+            )
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)